src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestRetryFailedManagedIndex.kt

```diff
@@ -0,0 +1,168 @@
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.ManagedIndexMetaData
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.transport.action.updateindexmetadata.UpdateManagedIndexMetaDataAction
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.transport.action.updateindexmetadata.UpdateManagedIndexMetaDataRequest
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.admin.cluster.state.ClusterStateRequest
+import org.elasticsearch.action.admin.cluster.state.ClusterStateResponse
+import org.elasticsearch.action.support.IndicesOptions
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.node.NodeClient
+import org.elasticsearch.cluster.ClusterState
+import org.elasticsearch.cluster.block.ClusterBlockException
+import org.elasticsearch.common.Strings
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentHelper
+import org.elasticsearch.index.Index
+import org.elasticsearch.rest.BaseRestHandler
+import org.elasticsearch.rest.BytesRestResponse
+import org.elasticsearch.rest.RestChannel
+import org.elasticsearch.rest.RestController
+import org.elasticsearch.rest.RestRequest
+import org.elasticsearch.rest.RestResponse
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.rest.action.RestActionListener
+import org.elasticsearch.rest.action.RestResponseListener
+
+class RestRetryFailedManagedIndex(
+    settings: Settings,
+    controller: RestController
+) : BaseRestHandler(settings) {
+    private val log = LogManager.getLogger(javaClass)
+
+    init {
+        controller.registerHandler(RestRequest.Method.POST, RETRY_BASE_URI, this)
+        controller.registerHandler(RestRequest.Method.POST, "$RETRY_BASE_URI/{index}", this)
+    }
+
+    override fun getName(): String {
+        return "retry_failed_managed_index"
+    }
+
+    override fun prepareRequest(request: RestRequest, client: NodeClient): RestChannelConsumer {
+        val indices: Array<String>? = Strings.splitStringByCommaToArray(request.param("index"))
+        if (indices == null || indices.isEmpty()) {
+            throw IllegalArgumentException("Missing indices")
+        }
+        val body = if (request.hasContent()) {
+            XContentHelper.convertToMap(request.requiredContent(), false, request.xContentType).v2()
+        } else {
+            mapOf()
+        }
+
+        val strictExpandIndicesOptions = IndicesOptions.strictExpand()
+
+        val clusterStateRequest = ClusterStateRequest()
+        clusterStateRequest.clear()
+            .indices(*indices)
+            .metaData(true)
+            .local(request.paramAsBoolean("local", clusterStateRequest.local()))
+            .masterNodeTimeout(request.paramAsTime("master_timeout", clusterStateRequest.masterNodeTimeout()))
```

Comment:
Do we expect this to take a long time and need the ability to increase the masterNodeTimeout?